.github/workflows/main.yaml

```diff
@@ -171,22 +181,31 @@ jobs:
       - name: Test
         run: |
           make test
-        if: matrix.target != 'aarch64-apple-darwin'
+        if: matrix.target != 'aarch64-apple-darwin' && matrix.build != 'linux-musl-x64'
+      - name: Test (musl)
+        if: matrix.build == 'linux-musl-x64'
+        run: |
+          LIBC=musl WASMER_CAPI_USE_SYSTEM_LIBFFI=1 make test
       - name: Test C API
         run: |
           make test-capi
-        if: matrix.os != 'windows-latest' && matrix.target != 'aarch64-apple-darwin' # we can't test yet on Apple Silicon or Windows
+        if: matrix.os != 'windows-latest' && matrix.target != 'aarch64-apple-darwin' && matrix.build != 'linux-musl-x64' # we can't test yet on Apple Silicon or Windows
```

Comment:
At some point we may just want to encode the logic in the matrix directly as a variable like `should-test-capi` and have `true`/`false` for each target.

I'll leave it up to your judgement what you think is better, just thinking out loud

Reply:
@MarkMcCaskey I totally agree.

I started my PR with much more things in the workflow/makefile (improvements, optimizations) but ran into problem and reverted everything to try to find the root cause.

I think we can explore this in a dedicated PR.

Reply:
I don't understand. We don't test the C API with musl, but we build it (just below this line)? Would be nice to test it, don't you think :-p?